internal/sql/repository/pipelineConfig/PipelineRepository.go

```diff
@@ -173,6 +175,21 @@ func (impl PipelineRepositoryImpl) FindByCiPipelineId(ciPipelineId int) (pipelin
 	}
 	return pipelines, nil
 }
+func (impl PipelineRepositoryImpl) FindByParentCiPipelineId(ciPipelineId int) (pipelines []*Pipeline, err error) {
+	err = impl.dbConnection.Model(&pipelines).
+		Column("pipeline.*").
+		Join("INNER JOIN app_workflow_mapping awm on awm.component_id = pipeline.id").
```

Comment:
check if it has index

Reply:
this query will only return cd pipeline which are immediate child of the ci pipeline, is that the intent? name doesnt say that also we could have queried workflow directly with parentId and parentType

Reply:
Yes, this method was created for getting immediate children of CI for handling successful CI events. And pipeline object was needed for methods used next in the logic, that's why I did not use parentId and parentType. I agree that the name is misleading. Is "FindImmediateCiChild" a better name?